### PR TITLE
Minor make adjustments, fix warnings

### PR DESCRIPTION
**Pull request by cjolivier01**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -79,7 +73,7 @@ data.o: src/data.cc
 recordio.o: src/recordio.cc
 config.o: src/config.cc
 
-$(LIB_DIR)/libdmlc.a: $(OBJ)
```
Comment (piiswrong) [inline Makefile]: why removing these?

File: src/io/s3_filesys.cc [hunk 2]
```diff
@@ -824,10 +824,10 @@ void ListObjects(const URI &path,
 S3FileSystem::S3FileSystem() {
   const char *keyid = getenv("AWS_ACCESS_KEY_ID");
   const char *seckey = getenv("AWS_SECRET_ACCESS_KEY");
-  if (keyid == NULL) {
+  if (keyid == NULL || !*keyid) {
     LOG(FATAL) << "Need to set enviroment variable AWS_ACCESS_KEY_ID to use S3";
   }
-  if (seckey == NULL) {
+  if (seckey == NULL || !*seckey) {
```
Comment (piiswrong) [inline src/io/s3_filesys.cc]: @mli Is this correct?

Comment (piiswrong) [inline src/io/s3_filesys.cc]: @mli is this correct?

Comment (cjolivier01) [inline src/io/s3_filesys.cc]: My implementation returns NULL (Ubuntu). However, the documentation is not specific on this and states that it returns NULL if the environment variable is not found.  However, I can put an empty variable in the environment:
[coolivie@DEV:~]export FOO=
[coolivie@DEV:~]env | grep FOO
FOO=
[coolivie@DEV:~]

Comment (cjolivier01) [inline src/io/s3_filesys.cc]: By googling that, I see cases of some implementations returning an empty string.

Also, this example does not appear to check:  
http://cvsweb.netbsd.org/bsdweb.cgi/src/lib/libc/stdlib/getenv.c?rev=1.19&content-type=text/x-cvsweb-markup

Comment (piiswrong) [inline src/io/s3_filesys.cc]: ok I'll merge this since it looks like it wouldn't hurt.

**Pull request merged**